developers/weaviate/client-libraries/python/async.md

```diff
@@ -0,0 +1,196 @@
+
+## Sync and async methods
+
+The async client object is designed to be used in an `async` function or in an [`asyncio` event loop](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio-event-loop).
+
+Accordingly, a majority of the client methods are `async` functions that return [`Coroutines` objects](https://docs.python.org/3/library/asyncio-task.html#coroutine). However, some methods are synchronous and can be used in a synchronous context.
+
+As a rule of thumb, a method that involves a request to Weaviate is likely to be an async function, while a method that executes in a local context is likely to be synchronous.
+
+### How to identify async methods
+
+Async methods are identified by its method signature. Async methods are defined with the `async` keyword, and they return a `Coroutines` object.
+
+To see a method signature, you can use the `help()` function in Python, or use an IDE that supports code completion such as [Visual Studio Code](https://code.visualstudio.com/docs) or [PyCharm](https://www.jetbrains.com/help/pycharm/viewing-reference-information.html).
+
+### Example async methods
+
+Methods that involve a request to Weaviate is likely to be an async function. For example, each of the following operations is an async function:
+
+- `async_client.connect()`: Connect to a Weaviate server
+- `async_client.collections.create()`: Create a new collection
+- `<collection_object>.data.insert_many()`: Insert a list of objects into a collection
+
+### Example sync methods
+
+Methods that execute in a local context are likely to be synchronous. For example, each of the following operations is a sync function:
+
+- `async_client.collections.get("<COLLECTION_NAME>")`: Create a Python object to interact with an existing collection (this does not create a collection)
+- `async_client.is_connected()`: Check the last known connection status to the Weaviate server
+
+## Context manager
+
+The async client can be used in an asynchronous context manager, in a pattern similar to:
+
+<FilteredTextBlock
+  text={PythonCode}
+  startMarker="# START AsyncContextManager"
+  endMarker="# END AsyncContextManager"
+  language="py"
+/>
+
+When using the async client in a context manager, you do not need to call `.connect()` or `.close()` explicitly. The client handles the connection and disconnection automatically.
+
+## Async usage examples
+
+The async client object largely provides the same functionality as the [synchronous Python client](./index.md), with some key differences. First, the async client is designed to be used in an `async` function or in an [`asyncio` event loop](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio-event-loop). Accordingly, many of the client methods are `async` functions that return [`Coroutines` objects](https://docs.python.org/3/library/asyncio-task.html#coroutine).
+
+To execute an async client method, you must `await` the method call or use `asyncio.run()` to run the method in an event loop.
```

Comment:
This is slightly inaccurate. I would instead say:

- To execute an async client method, you must `await` it inside another `async` function. To execute an `async` function in your Python script, you can use `asyncio.run(my_async_function)` or the event loop directly:
```python
loop = asyncio.new_event_loop()
loop.run_until_complete(my_async_function())
```